auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/GrpcUmsClient.java

```diff
@@ -371,6 +371,35 @@ public MachineUser getMachineUserDetails(String userCrn, String accountId, Optio
         }
     }
 
+    /**
+     * Get or Create new machine user for given machineUserName.
+     *
+     * @param machineUserName new machine user name
+     * @param accountId       the accountId
+     * @param requestId       an optional request Id
+     * @return the machineUser
+     */
+    @Retryable(value = UmsOperationException.class, maxAttempts = 10, backoff = @Backoff(delay = 5000))
+    public MachineUser getOrCreateMachineUserWithoutAccessKey(String machineUserName, String accountId, Optional<String> requestId) {
+        try {
+            UmsClient client = makeClient(channelWrapper.getChannel());
+            String generatedRequestId = RequestIdUtil.getOrGenerate(requestId);
+            LOGGER.debug("Creating machine user {} for accountId {} using request ID {}", machineUserName, accountId, generatedRequestId);
+            MachineUser machineUser = client.getOrCreateMachineUserWithoutAccessKey(generatedRequestId, accountId, machineUserName);
+            LOGGER.debug("Machine User retrieved for machineUserName: {}, machineUser: {}", machineUserName, machineUser);
+            return machineUser;
+        } catch (StatusRuntimeException ex) {
+            if (Status.UNAVAILABLE.getCode().equals(ex.getStatus().getCode())) {
+                String errMessage = String.format("Cannot create machinue user '%s' for '%s' as " +
+                        "UMS API is UNAVAILABLE at the moment", machineUserName, accountId);
+                LOGGER.debug(errMessage, ex);
+                throw new UmsOperationException(errMessage, ex);
+            } else {
+                throw ex;
+            }
+        }
+    }
+
```

Comment:
what is the difference between this and method `createMachineUser` ? As I see there are common parts, maybe it is worth to create different method for common parts, what do you think?

Reply:
@aakashshankar can you check this comment of mine?